server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -3469,8 +3469,31 @@ private void doSnapshotShard(SnapshotShardContext context) {
     private static void ensureNotAborted(ShardId shardId, SnapshotId snapshotId, IndexShardSnapshotStatus snapshotStatus, String fileName) {
         try {
             snapshotStatus.ensureNotAborted();
+
+            if (snapshotStatus.getStage() != IndexShardSnapshotStatus.Stage.INIT
+                && snapshotStatus.getStage() != IndexShardSnapshotStatus.Stage.STARTED) {
+                // A normally running shard snapshot should be in stage INIT or STARTED. And we know it's not in PAUSING or ABORTED because
+                // the ensureNotAborted() call above did not throw. The remaining options don't make sense, if they ever happen.
+                logger.info(
+                    "Shard snapshot found no interrupt signal, but is in an unexpected state. ShardId [{}], SnapshotID [{}], Stage [{}]",
+                    shardId,
+                    snapshotId,
+                    snapshotStatus.getStage()
+                );
+            }
         } catch (Exception e) {
-            logger.debug("[{}] [{}] {} on the file [{}], exiting", shardId, snapshotId, e.getMessage(), fileName);
+            // We want to see when a shard snapshot operation checks for and finds an interrupt signal during shutdown. A
+            // PausedSnapshotException indicates we're in shutdown because that's the only case when shard snapshots are signaled to pause.
+            // An AbortedSnapshotException may also occur during shutdown if an uncommon error occurs.
+            logger.info(
```

Comment:
One thought that comes to mind is that I could add a static map that uniquely identifies callers by shardId-snapshotId. Then we could reduce the logging frequency however we wished. We'd have to cleanup or reset the map, which could be done with an extra mechanism, or timers, or by limiting the size of the map and having a recency bias.

Reply:
I think we don't want to do this at `info` by default. I'd suggest `debug` and then enabling the appropriate `debug` logger by default in serverless. If you'd rather not use the top-level `BlobStoreRepository` logger then you can make a new subsidiary one (e.g. make a new dummy inner class with its own logger)

Reply:
Done.